lib/Agrammon/Outputs/FilterGroupCollection.pm6

```diff
@@ -112,6 +112,29 @@ class Agrammon::Outputs::FilterGroupCollection {
         Agrammon::Outputs::FilterGroupCollection.new(instances => @result-instances)
     }
 
+    #| Check in the other filter group collection for values that are greater than a threshold
+    #| and return a new group collection as the result. When $push-all is False, then the method
+    #| returns only filter groups where $thresh is exceeded. Otherwise, if $push-all is True,
+    #| filter groups where $thresh is not exceeded are also returned with a value of $thresh.
+    method select-by-threshold(Agrammon::Outputs::FilterGroupCollection $other, Numeric $thresh, Bool $push-all) {
+        # Only process those groups existing on the other side.
+        my @result-instances;
+        for $other!internal-values-by-filter -> $their-elem {
+            my $their-key = $their-elem.key;
+            if $their-elem.value > $thresh {
+                with %!values-by-filter{$their-key} -> $our-value {
+                    # Push our value if our value > threshold
+                    @result-instances.push: $their-key => $our-value;
+                }
+            } elsif $push-all {
+                # Push threshold if our value <= threshold
+                @result-instances.push: $their-key => $thresh;                
```

Comment:
This branch doesn't do a `with %!values-by-filter{$their-key}`, meaning that if the key does not exist in our filter group, but does exist in the other filter group, then we'll include `$thresh` for it anyway. Is this correct? There's no test that seems to cover this case. Such a test would be like:

```
    my $group-a = Agrammon::Outputs::FilterGroupCollection.from-filter-to-value-pairs:
            [{ac => 'blue cow'} => 4, {ac => 'green cow'} => 31];
    my $group-b = Agrammon::Outputs::FilterGroupCollection.from-filter-to-value-pairs:
            [{ac => 'blue cow'} => 0, {ac => 'green cow'} => 5, {ac => 'pink cow'} => 9];
    given $group-a.select-by-threshold($group-b, 5, True) {
        # ...correct checks here...
    }
```

In this example, the question is if the result should contain a filter key involving pink cow or not? I believe it will, from looking at the code, but I'm not sure - based on the function description - if it should.

Reply:
(And whichever way it is, it's good to test this case.)

Reply:
Yes it should contain pink cow. Will add the test.

Reply:
@jnthn What is the value of `%!values-by-filter{$their-key}` if the key is missing in ours? Is it 0? or not defined?

Reply:
@ChHaeni An undefined value (one that will give `False` in response to `.defined`; `with $value { }` is like `if $value.defined { }`, except it also topicalizes on `$value` (sets the default variable `$_` inside of the block to `$value`).